Source/CombatExtended/CombatExtended/Verbs/Verb_ShootCE.cs

```diff
@@ -94,10 +94,42 @@ protected override float SwayAmplitude
         // Whether our shooter is currently under suppressive fire
         private bool IsSuppressed => ShooterPawn?.TryGetComp<CompSuppressable>()?.isSuppressed ?? false;
 
+        //Checks if magazine-fed weapon needs to reload before able to shoot again (out-of-ammo state is automatically handled by TryStartReload)
```

Comment:
Why not use the properties already included in CompAmmo?

Reply:
If you mean why not use the CanBeFiredNow property: because that one always returns true when the ammo system is disabled, even though weapons still need to reload when their magazine goes empty. Having checked again, I still can't find a property that does exactly what I need for NeedsMagazineReload.

Reply:
There's a slew of properties for all combinations of magazine and ammo states that should make all these additions redundant.

Reply:
After the change to make TryStartReload() handle out-of-ammo state properly for bows/etc, I was able to simplify the code a fair bit, but I'm still having to use a large combination of AmmoUser properties to know whether TryStartReload() should be called. Here's my dilemma:

For magazine weapons, we just need to know if the mag count is at 0. It doesn't matter if there's ammo in the inventory, or if the ammo system is disabled entirely, the resulting action is always the same: call TryStartReload() so it either puts a fresh mag in, or unequips the weapon. In code terms, that's (HasMagazine && CurMagCount == 0).

For no-mag weapons, it's slightly different: We do care about ammo in the inventory, because the weapon feeds directly off it. But only if the ammo system is on. That translates to (!HasMagazine && UseAmmo && !HasAmmo).

In summary, a property is needed that resolves ((HasMagazine && CurMagCount == 0) || (!HasMagazine && UseAmmo && !HasAmmo)). Is there one in CompAmmoUser that does such a thing? No...  !CanBeFiredNow comes really close, but it assumes that magazines never need to reload when the ammo system is disabled, so it doesn't make the cut. (That might well be a bug for all I know, though)

So where do I go from here? Is the logic of CanBeFiredNow intentional, or do I need to fix it? If it's intended, should I make a separate new property in CompAmmoUser for the needed purpose?